File: 442. Find All Duplicates in an Array/_442.py
class Solution:
    def findDuplicates(self, nums):
        nums.sort()
        res = list()
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1]:
                res.append(nums[i])

        return res

    def findDuplicates1(self, nums):
        res = []
        for x in nums:
            if nums[abs(x)-1] < 0:
                res.append(abs(x))
            else:
                nums[abs(x)-1] *= -1
        return res

File: 442. Find All Duplicates in an Array/442.go
package leetcode

import(
	"sort"
)

func findDuplicates(nums []int) []int {
	sort.Ints(nums)
	res := make([]int, 0)
	for i:=0; i<len(nums)-1; i++{
		if nums[i] == nums[i+1]{
			res = append(res, nums[i])
		}
	}
	return res
}